src/commands/github/actionCommands.ts

```diff
@@ -34,4 +41,62 @@ export async function cancelAction(context: IActionContext, node?: ActionTreeIte
     const client: Octokit = await createOctokitClient();
     await client.actions.cancelWorkflowRun({ owner: node.data.repository.owner.login, repo: node.data.repository.name, run_id: node.data.id });
     await node.refresh(); // need to refresh to update the data
+    await checkActionStatus(context, node, 'cancel');
+}
+
+async function checkActionStatus(context: IActionContext, node: ActionTreeItem, action: 'rerun' | 'cancel'): Promise<void> {
+    // realistically, we have no idea how long someone's build can take, but might as well keep retrying while VS Code is opened
+    const tokenSource: CancellationTokenSource = new CancellationTokenSource();
+    const token: CancellationToken = tokenSource.token;
+    if (activeActionTokens[node.data.id]) {
+        activeActionTokens[node.data.id]?.cancel();
+    }
+
+    activeActionTokens[node.data.id] = tokenSource;
+
+    const retries: number = 10;
```

Comment:
I don't think I would use exponential backoff for something like this. I mostly use it to retry an action itself, not to get the status of that action. If you're retrying the action itself, you need the backoff otherwise you might exacerbate the problem on the server. We've talked about polling in the past and how I don't like auto-refresh in the tree, but this is the exact scenario where I think polling is fine (as long as you're careful). The polling is directly related to an action the user just took, and it's limited to a pretty small scope. This is very similar to other scenarios like creating an azure resource or deploying (except you have to manually do the polling instead of letting the sdk do it for us).

You might want to see if you can combine this with [pollAzureAsyncOperation](https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/utils/azureUtils.ts#L35) already in this repo.

Reply:
Hmm, I think the only caveat here though is that a build could potentially take hours.  While polling isn't that expensive of an operation, I feel like doing it every 10 seconds for 4 hours is kind of excessive.  Maybe I can build my own back-off where if it's past 10 minutes, it goes to polling every minute or something?  My expectation is most people will just leave it running and miss the notification, but expect it to be updated in the tree or they'll just close VS Code and see it's complete later.  What do you think?

Reply:
Yeah part of "as long as you're careful" is you have to pick a reasonable interval and timeout. Plan around what you actually expect to happen in the majority of cases, not the 4-hour edge-case. Ideally this involves running a bunch of static web app builds and seeing how long they take to pick your number. I would guess around 10 minutes would be plenty, but maaaybe up to 20? After that, I would just give up and say something like "The build is still running, check &lt;link to github&gt; for status".

Reply:
I figured I'd start at 20 and look at telemetry to see if we need to adjust the time (either up or down).  Based off telemetry, not that many people really use re-run action that anyway.